org.eclipse.ice.commands/src/main/java/org/eclipse/ice/commands/LocalCommand.java

```diff
@@ -115,13 +98,19 @@ protected CommandStatus run() {
 			status = setupProcessBuilder(thisCommand);
 
 			// Run the job
-			status = runProcessBuilder();
+			status = processJob();
+
+			// Finish the job by logging output. If the job is not completed, then
+			// it will be monitored in monitorJob
+			status = finishJob();
 
 			// Check the status to ensure the job hasn't failed
 			try {
 				checkStatus(status);
 			} catch (IOException e) {
+				// If not, return failed
 				e.printStackTrace();
```

Comment:
Should be logged or not.

Reply:
It is logged in checkStatus

Reply:
you should also remove all references to e.printStackTrace().

Reply:
also, if the exception is handled in checkStatus then it should not rethrow the exception only to be ignored.

Reply:
> you should also remove all references to e.printStackTrace().

Why should I do this? Isn't this useful for debugging purposes?

> also, if the exception is handled in checkStatus then it should not rethrow the exception only to be ignored.

checkStatus logs whether or not the status is failed and then throws an IOException, which is why if checkStatus is caught the run function returns CommandStatus.FAILED. The exception isn't handled in checkStatus, it is just logged, and then handled in the catch to checkStatus by returning that the job has failed.

Reply:
> > you should also remove all references to e.printStackTrace().
> 
> Why should I do this? Isn't this useful for debugging purposes?
> 

No, because you can't guarantee that stdout is going to go somewhere useful, and because it can't be controlled. Also, since you're ignoring the exception, the user is going to see this when something fails have no idea what is going on. It is better to always use a logging framework for debugging so it is only visible to the developer.

> > also, if the exception is handled in checkStatus then it should not rethrow the exception only to be ignored.
> 
> checkStatus logs whether or not the status is failed and then throws an IOException, which is why if checkStatus is caught the run function returns CommandStatus.FAILED. The exception isn't handled in checkStatus, it is just logged, and then handled in the catch to checkStatus by returning that the job has failed.

I'm not sure why you'd want to throw an exception at this point then, rather than just return a boolean. Particularly since you're just essentially ignoring the exception anyway.

Reply:
> > > also, if the exception is handled in checkStatus then it should not rethrow the exception only to be ignored.
> > 
> > 
> > checkStatus logs whether or not the status is failed and then throws an IOException, which is why if checkStatus is caught the run function returns CommandStatus.FAILED. The exception isn't handled in checkStatus, it is just logged, and then handled in the catch to checkStatus by returning that the job has failed.
> 
> I'm not sure why you'd want to throw an exception at this point then, rather than just return a boolean. Particularly since you're just essentially ignoring the exception anyway.

Okay, that is fair. I've just changed the function to a return a boolean, and if the status is bad the job is marked as failed.

> > > you should also remove all references to e.printStackTrace().
> > 
> > 
> > Why should I do this? Isn't this useful for debugging purposes?
> 
> No, because you can't guarantee that stdout is going to go somewhere useful, and because it can't be controlled. Also, since you're ignoring the exception, the user is going to see this when something fails have no idea what is going on. It is better to always use a logging framework for debugging so it is only visible to the developer.

Okay, I have gone through and removed all instances where an exception is thrown to print the stack trace.

> > > also, if the exception is handled in checkStatus then it should not rethrow the exception only to be ignored.
> > 
> > 
> > checkStatus logs whether or not the status is failed and then throws an IOException, which is why if checkStatus is caught the run function returns CommandStatus.FAILED. The exception isn't handled in checkStatus, it is just logged, and then handled in the catch to checkStatus by returning that the job has failed.
> 
> I'm not sure why you'd want to throw an exception at this point then, rather than just return a boolean. Particularly since you're just essentially ignoring the exception anyway.